src/providers/pkcs11_provider/utils.rs

```diff
@@ -258,3 +417,134 @@ impl Drop for Session<'_> {
         }
     }
 }
+
+/// This function converts key parameters from Parsec flavour to PKCS11 flavour
+///
+/// The return type is a tuple with:
+/// * mechanism for key generation
+/// * attributes list for public key
+/// * attributes list for private key
+/// * allowed mechanism for the two halves of the key
+pub fn parsec_to_pkcs11_params(
+    attributes: Attributes,
+    key_id: &[u8],
+) -> Result<(
+    CK_MECHANISM,
+    Vec<CK_ATTRIBUTE>,
+    Vec<CK_ATTRIBUTE>,
+    CK_MECHANISM_TYPE,
+)> {
+    match attributes.key_type {
+        Type::RsaKeyPair => {
+            let mechanism = CK_MECHANISM {
+                mechanism: CKM_RSA_PKCS_KEY_PAIR_GEN,
+                pParameter: std::ptr::null_mut(),
+                ulParameterLen: 0,
+            };
```

Comment:
I wonder if it's worth expanding this to support for ECDSA to have parity with the TPM provider, but figuring out which attributes go where might be a pain.